apollo-client/src/main/java/com/ctrip/framework/apollo/spring/annotation/ApolloAnnotationProcessor.java

```diff
@@ -244,15 +251,22 @@ private Object resolvePropertyValue(String beanName, String placeHolder) {
     return propertyValue;
   }
 
-  private Object parseJsonValue(String json, Type targetType) {
+  private Object parseJsonValue(String json, Type targetType, String datePattern) {
     try {
-      return GSON.fromJson(json, targetType);
+      return DATEPATTERN_GSON_MAP.computeIfAbsent(datePattern, this::buildGson).fromJson(json, targetType);
```

Comment:
Is the `computeIfAbsent` method of `HashMap` thread safe?

Reply:
Regarding the `HashMap`, `computeIfAbsent` is not thread-safe. If needed, we can use a `ConcurrentHashMap`. However, is there intense thread contention in `AutoUpdateConfigChangeListener`? Is it necessary to make any change?

Reply:
I think there would be concurrent execution situations.

Reply:
I've already changed it to a `ConcurrentHashMap`.  :smiley: